daemon/daemon_test.go

```diff
@@ -335,6 +337,54 @@ func TestDaemonReloadLabels(t *testing.T) {
 	}
 }
 
+func TestDaemonReloadInsecureRegistries(t *testing.T) {
+	daemon := &Daemon{}
+	daemon.RegistryService = registry.NewService(registry.ServiceOptions{})
+	daemon.configStore = &Config{}
```

Comment:
I agree with @cpuguy83's suggestion and don't see anything in this test that removes insecure registries from the config.

Looking at the code, I don't think this is supported though. I would expect it to be possible to remove insecure-registries entries by reloading the config.